### DISCUSSION TITLE
Dynamic routes for static `.pdf` files

### DISCUSSION THREAD

**Discussion by shelegdmitriy** [near/near-discovery#1002] [Q&A] [answered]
There are several `.pdf` documents available at the `/papers` route. To make it work we have them all inside of `/public/files/` folder and then pass them as [props](https://github.com/near/near-discovery/blob/develop/src/pages/papers/index.tsx#L17-L19) to the components repo. This is not the best way of doing that I think.

The way it can be improved we could add a dynamic route to the `next.config.js` file like so:

```
const nextConfig = {
// config params
  redirects: async () => {
      return [
        // available routes
        {
          source: '/papers/:slug',
          destination: '/files/:slug.pdf',
          permanent: true,
        },
        // other routes
      ]
  },
// next config params
}
```

**Pros:**
- easy access to the file across near.org;
- no need to pass files as props to components;

**Cons:**
- file names should be named using [kebab-case](https://developer.mozilla.org/en-US/docs/Glossary/Kebab_case);
- file names should be pretty short and clear;

**Comment by charleslavon** [comment, marked as answer]
hey @shelegdmitriy thanks for starting a discussion. I plan to look into leveraging a cdn to serve these files, to reduce the amount of bandwith from our vercal plan that we're using to serve these files because these pdfs have become the top sources of bandwidth consumption on vercel.   So there will likely be a change in the URLs we use to reference these PDFs. 

Can you clarify what problem you're trying to solve? Is it how to avoid passing props which contain the PDF titles into bos components?  We could also hard code them within components like we do our relative urls `/papers/the-white-paper`.

**Comment by shelegdmitriy** [comment]
> Is it how to avoid passing props which contain the PDF titles into bos components?

Mainly, Yes. Besides I saw there is a route for the **nightshade** (`/papers/nightshade`) so I thought this dynamic routing for docs this this might be helpful.

> I plan to look into leveraging a cdn to serve these files, to reduce the amount of bandwith from our vercal plan that we're using to serve these files because these pdfs have become the top sources of bandwidth consumption on vercel. So there will likely be a change in the URLs we use to reference these PDFs.

Ok, then no action needed in this discussion. Thanks @charleslavon!
